spec/operatoroverloading.dd

```diff
@@ -487,7 +487,38 @@ $(H3 $(LNAME2 compare, Overloading $(D <), $(D <)$(D =), $(D >), and $(D >)$(D =
         If they resolve to different functions, the best matching one is used.
         If they both match the same, but are different functions, an ambiguity
         error results.)
+---
+struct B
+{
+    int opCmp(int)         { return -1; }
+    int opCmp(ref const S) { return -1; }
+    int opCmp(ref const C) { return -1; }
+}
+
+struct S
+{
+    int opCmp(ref const S) { return 1; }
+    int opCmp(ref B)       { return 0; }
+}
 
+struct C
+{
+    int opCmp(ref const B) { return 0; }
+}
+
+void main()
+{
+    S s;
+    const S cs;
+    B b;
+    C c;
+    assert(s > s);      // s.opCmp(s) > 0
+    assert(!(s < b));   // s.opCmp(b) > 0  - S.opCmp(ref B) is exact match
+    assert(!(b < s));   // s.opCmp(b) < 0  - S.opCmp(ref B) is exact match
+    assert(b < cs);     // b.opCmp(s) < 0  - B.opCmp(ref const S) is  exact match
+    static assert(!__traits(compiles, b < c)); // both C.opCmp and B.opcmp match exactly
```

Comment:
Note that, due to a bug, this assert will fail.

Reply:
Is it in bugzilla?

Reply:
https://issues.dlang.org/show_bug.cgi?id=21453